### ISSUE TITLE
SNOW-449714: PUT/GET issue with version 2.6.0 in certain deployments

### ISSUE THREAD

**Issue by c85** [snowflakedb/snowflake-connector-python#850]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

   `python version 3.9.0`

2. What operating system and processor architecture are you using?

  N/A - `Microsoft Azure Function App`

3. What are the component versions in the environment (`pip freeze`)?

   `snowflake-connector-python` version 2.6.0

4. What did you do?

   When deploying the function app to Microsoft Azure with the `snowflake-connector-python` v.2.6.0 component, it began to throw an HTTPError when using the write_pandas function. I noticed specifically it began to fail at the COPY step with the following error:

`400 Client Error: Bad Request for url`

5. What did you expect to see?

   No error or warning

6. Can you set logging to DEBUG and collect the logs?

   ```
Result: Failure Exception: TypeError: __str__ returned non-string (type HTTPError) Stack
400 Client Error: Bad Request for url: https://customer-stage.s3.amazonaws.com/.../file0.txt
   ```

**Comment by sfc-gh-hkapre** [comment]
@c85 What cloud/deployment is your Snowflake account running on? 

We are aware of an issue with PUT commands for some deployments with our latest connector. This was introduced as part of our efforts to bypass/remove the cloud sdk dependencies for PUT/GET. Can you try setting this parameter which will revert to the old code path?

use_new_put_get=False in the connect() method.

**Comment by c85** [comment]
> @c85 What cloud/deployment is your Snowflake account running on?
> 
> We are aware of an issue with PUT commands for some deployments with our latest connector. This was introduced as part of our efforts to bypass/remove the cloud sdk dependencies for PUT/GET. Can you try setting this parameter which will revert to the old code path?
> 
> use_new_put_get=False in the connect() method.

Thank you @sfc-gh-hkapre I added the new connection parameter and it works! For reference, the cloud for our Snowflake account is AWS.

**Comment by sfc-gh-mkeller** [comment]
@c85 I want to make sure that this is resolved by #846 instead of just having the new code path being turned off for you.
Could you please tell me which Region your account is in? I attached a screenshot where you can see this info on the UI once you log in.
<img width="314" alt="Screen Shot 2021-09-02 at 3 29 10 PM" src="https://user-images.githubusercontent.com/63477823/131924533-87ec1090-0e76-47d7-80f6-c9aede81d5bf.png">

**Comment by c85** [comment]
> @c85 I want to make sure that this is resolved by #846 instead of just having the new code path being turned off for you.
> Could you please tell me which Region your account is in? I attached a screenshot where you can see this info on the UI once you log in.
> <img alt="Screen Shot 2021-09-02 at 3 29 10 PM" width="314" src="https://user-images.githubusercontent.com/63477823/131924533-87ec1090-0e76-47d7-80f6-c9aede81d5bf.png">

Region: US East (Ohio)

**Comment by sfc-gh-mkeller** [comment]
I just manually tested this @c85 the next release (`2.6.1`) will work for your deployment. I appologize for the inconvenience and thank you for reporting this issue!
In the meantime there are a couple options to work around the issue:
1. You can downgrade to `2.5.1`
2. You can supply `use_new_put_get=False` to your connection object.
   Similar to:
   ```python
   with snowflake.connector.connect(
    account="", user="", ...,
    use_new_put_get=False,
    )
   ```
3. You could install the fix for this bug from Git, please be aware that this will require some sort of compiler so that Cython can compile our native code. The process should just work with the following command: `python -m pip install git+https://github.com/snowflakedb/snowflake-connector-python.git@mkeller/SNOW-443844-put_wrong_url#egg=snowflake-connector-python`

**Comment by sfc-gh-mkeller** [comment]
For anyone new who finds this issue the following AWS deployments are all affected by the bug that's being resolved in #846 :
> The Europe (Frankfurt), US East (Ohio), Canada (Central), Europe (London), Asia Pacific (Seoul), Asia Pacific (Mumbai), Europe (Paris), China (Ningxia), Europe (Stockholm), Asia Pacific (Osaka Local), AWS GovCloud (US-East), and Asia Pacific (Hong Kong) Regions were launched after 2013, and support SigV4 but not SigV2. If you have code that accesses S3 buckets in that region, it is already making exclusive use of SigV4.

**Comment by sfc-gh-mkeller** [comment]
Any affected users should try installing the newly released `2.6.1` [connector](https://pypi.org/project/snowflake-connector-python/2.6.1/) and try removing the `use_new_put_get=False` workaround flag to see if the new version has fixed the issue.
We apologize again for any inconvenience and are looking forward to your feedback!
